bundle/src/main/java/com/adobe/acs/commons/indesign/dynamicdeckdynamo/models/DynamicDeckInitiatorPageModel.java

```diff
@@ -0,0 +1,154 @@
+package com.adobe.acs.commons.indesign.dynamicdeckdynamo.models;
+
+import com.adobe.acs.commons.mcp.form.*;
+import org.apache.sling.api.SlingHttpServletRequest;
+import org.apache.sling.api.resource.Resource;
+import org.apache.sling.models.annotations.DefaultInjectionStrategy;
+import org.apache.sling.models.annotations.Model;
+
+import javax.inject.Inject;
+import javax.inject.Named;
+
+@Model(
+        adaptables = {Resource.class, SlingHttpServletRequest.class},
+        defaultInjectionStrategy = DefaultInjectionStrategy.OPTIONAL)
+public class DynamicDeckInitiatorPageModel extends GeneratedDialog {
+
+    @Inject
+    @Named(value = "deckTitle")
+    @FormField(
+            name = "Deck Title",
+            required = true,
+            category = "General",
+            description = "Enter a InDesign deck name. e.g. summer-2020",
+            hint = "sample")
+    private String deckTitle;
+
+    @Inject
+    @Named(value = "destinationPath")
+    @FormField(
+            name = "Deck Destination Folder Path",
+            description = "Select the folder path where the InDesign deck will be generated.",
+            required = true,
+            component = PathfieldComponent.FolderSelectComponent.class,
+            options = {"base=/content/dam"},
+            hint = "/content/dam/someFolder",
+            category = "General")
+    private String destinationPath;
+
+    @Inject
+    @Named(value = "masterAssetPath")
+    @FormField(
+            name = "Master Asset Path",
+            description = "Select the master asset. InDesign deck's header & footer info will be taken from master asset's metadata",
+            component = PathfieldComponent.AssetSelectComponent.class,
+            options = {"base=/content/dam"},
+            hint = "/content/dam/someAsset",
+            category = "General")
+    private String masterAssetPath;
+
+    @Inject
+    @Named(value = "templatePath")
+    @FormField(
+            name = "Template Folder Path",
+            description = "Select the InDesign template folder for generating the InDesign deck. Make sure that the folder must have a template file and the mapped XML file in the same folder.",
+            required = true,
+            component = PathfieldComponent.FolderSelectComponent.class,
+            options = {"base=/content/dam"},
+            hint = "/content/dam/someFolder",
+            category = "General")
+    private String templatePath;
+
+    public enum Mode {
+        COLLECTION, QUERY, TAGS
+    }
+
+    @Inject
```

Comment:
Can we use an explicit Injector here? Ex. `@ValueMapValue` if it comes from the resource property?